compiler/src/dotty/tools/dotc/core/ContextOps.scala

```diff
@@ -55,11 +55,18 @@ object ContextOps:
     final def javaFindMember(name: Name, pre: Type, lookInCompanion: Boolean, required: FlagSet = EmptyFlags, excluded: FlagSet = EmptyFlags): Denotation =
       assert(ctx.isJava)
       inContext(ctx) {
-
+        import dotty.tools.dotc.core.NameOps.*
         val preSym = pre.typeSymbol
 
         // 1. Try to search in current type and parents.
-        val directSearch = pre.findMember(name, pre, required, excluded)
+        val directSearch =
+          if name.isTypeName && name.endsWith(StdNames.str.MODULE_SUFFIX) then
+            pre.findMember(name.stripModuleClassSuffix, pre, required, excluded) match
```

Comment:
It searches for `<root>.Foo$` when I replace `private static final example.Foo$` with `import example.Foo$` and `private static final Foo$`, and results in missing symbol.

Reply:
NOTE: importing companion object from Java works.

The following example compiles.

```java
package example;

import example.Foo$;

public class Bar {
	private static final example.Foo$ MOD = example.Foo$.MODULE$;
}
```

But the following does not.

```java
package example;

import example.Foo$;

public class Bar {
	private static final Foo$ MOD = Foo$.MODULE$;
}
```

Reply:
There is a different code path fro a raw identifier like `Foo$` look in `findRef`

Reply:
Thank you!